core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -508,4 +515,76 @@ private void updateBlockInStream(long blockId) throws IOException {
       mShouldCacheCurrentBlock = mAlluxioStorageType.isStore();
     }
   }
+
+  /**
+   * Seeks to a file position. Blocks are not cached unless they are fully read. This is only called
+   * by {@link FileInStream#seek}.
+   *
+   * @param pos The position to seek to. It is guaranteed to be valid (pos >= 0 && pos != mPos &&
+   *            pos <= mFileLength)
+   * @throws IOException if the seek fails due to an error accessing the stream at the position
+   */
+  private void seekInternal(long pos) throws IOException {
+    seekBlockInStream(pos);
+    checkAndAdvanceBlockInStream();
+    mCurrentBlockInStream.seek(mPos % mBlockSize);
+  }
+
+  /**
+   * Seeks to a file position. Blocks are cached even if they are not fully read. This is only
+   * called by {@link FileInStream#seek}.
```

Comment:
no. I am not changing skip().  No one is using it for now.  We can change it later when it is used.